src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,41 +1,73 @@
+import { useDispatch } from 'react-redux';
+import { getUser } from '../../api';
+import { setCurrentTodo } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const currentTodo = selector(state => state.currentTodo);
+  const [loader, setLoader] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    setLoader(true);
+
+    if (currentTodo) {
+      getUser(currentTodo.userId)
+        .then(setUser)
+        .finally(() => setLoader(false));
+    }
+  }, [currentTodo, dispatch]);
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {loader && <Loader />}
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+      {!loader && (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              Todo #{currentTodo?.id}
+            </div>
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => dispatch(setCurrentTodo(null))}
+            />
+          </header>
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+          <div className="modal-card-body">
+            <p className="block" data-cy="modal-title">
+              {currentTodo?.title}
+            </p>
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+            <p className="block" data-cy="modal-user">
+              {/* For not completed */}
+              {!currentTodo?.completed && (
+                <strong className="has-text-danger">Planned</strong>
+              )}
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+              {/* For completed */}
+              {currentTodo?.completed && (
+                <strong className="has-text-success">Done</strong>
+              )}
+              {' by '}
+              <a href={`mailto:${user?.email}`}>{user?.name}</a>
```

Comment:
Ensure that `user` is not null before accessing its properties like `email` and `name`. Consider adding a conditional check to prevent potential null reference errors.